ql/src/java/org/apache/hadoop/hive/ql/exec/tez/DagUtils.java

```diff
@@ -1600,9 +1601,14 @@ public Vertex createVertex(JobConf conf, BaseWork workUnit, Path scratchDir,
     // final vertices need to have at least one output
     boolean endVertex = tezWork.getLeaves().contains(workUnit);
     if (endVertex) {
+      OutputCommitterDescriptor ocd = null;
+      if (conf.get("hive.tez.mapreduce.output.committer.class") != null
+          && conf.get("mapred.output.committer.class") != null) {
```

Comment:
why is the presence of "mapred.output.committer.class" is needed?

Reply:
After thinking about it, we shouldn't have this check. Just because `MROutputCommitter` delegates all the calls to an inner committer object (implementation of this inner field is determined by `mapred.output.committer.class`), it doesn't mean all Tez output committer implementations should or will in the future.

Reply:
thanks!